mediorum/server/placement.go

```diff
@@ -46,17 +48,31 @@ func NewRendezvousHasher(hosts []string) *RendezvousHasher {
 	deadHosts := "https://content.grassfed.network/"
 	liveHosts := make([]string, 0, len(hosts))
 	for _, h := range hosts {
-		if !strings.Contains(deadHosts, h) {
-			liveHosts = append(liveHosts, h)
+		// dead host
+		if strings.Contains(deadHosts, h) {
+			continue
 		}
+
+		// invalid url
```

Comment:
this feels a bit like a bandaid, do we know the root cause yet?

Reply:
it is... I think the mutex is the real fix.